demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -628,90 +1002,178 @@ int main(int argc, char* argv[]) {
                 ie.SetConfig({{PluginConfigParams::KEY_DYN_BATCH_ENABLED, PluginConfigParams::YES}}, "GPU");
             }
 
-            if (FLAGS_pc)
-                ie.SetConfig({{PluginConfigParams::KEY_PERF_COUNT, PluginConfigParams::YES}});
-
             loadedDevices.insert(device);
         }
 
-        std::unique_ptr<AsyncDetection<DetectedAction>> action_detector;
+        // NOTE: Empty strings instead empty object of Params class 
+        cv::gapi::ie::Params<custom::FaceDetector> det_net("","","");
+        cv::gapi::ie::Params<custom::LandmarksDetector> landm_net("","","");
+        cv::gapi::ie::Params<custom::FaceReidentificator> reident_net("","","");
+        cv::gapi::ie::Params<custom::PersonDetActionRec> action_net("", "", "");
+
         if (!ad_model_path.empty()) {
             // Load action detector
-            ActionDetectorConfig action_config(ad_model_path);
-            action_config.deviceName = FLAGS_d_act;
-            action_config.ie = ie;
-            action_config.is_async = true;
-            action_config.detection_confidence_threshold = static_cast<float>(FLAGS_t_ad);
-            action_config.action_confidence_threshold = static_cast<float>(FLAGS_t_ar);
-            action_config.num_action_classes = actions_map.size();
-            action_detector.reset(new ActionDetection(action_config));
-        } else {
-            action_detector.reset(new NullDetection<DetectedAction>);
+            config::action_config.new_network = util::IsNetworkNewVersion(ad_model_path);
+            config::action_config.detection_confidence_threshold = static_cast<float>(FLAGS_t_ad);
+            config::action_config.action_confidence_threshold = static_cast<float>(FLAGS_t_ar);
+            config::action_config.num_action_classes = actions_map.size();
+
+            std::array<std::string, 7> outputBlobList;
+            !config::action_config.new_network ? outputBlobList = {"mbox_loc1/out/conv/flat",
```

Comment:
Probably we should only support _new_ networks in our sample?

Reply:
These networks recognize different amounts of actions.